hack/deploy.sh

```diff
@@ -0,0 +1,221 @@
+	.spec.startingCSV = env(OPERATOR_PKG_NAME) |
+	.spec.name = env(OPERATOR_PKG_NAME) |
+	.spec.source = env(OPERATOR_PKG_NAME)
+' "${SUBSCRIPTION_PATH}"
+oc apply -n "${NAMESPACE}" -f "${SUBSCRIPTION_PATH}"
+
+info 'waiting for CSV...'
+
+for n in {1..60}; do
+	phase="$(oc get csv "${OPERATOR_PKG_NAME}" -n "${NAMESPACE}" -o jsonpath='{.status.phase}' 2> /dev/null)"
+	if [ ! "$phase" = 'Succeeded' ]; then
+		if [ "${n}" = "60" ]; then
+			error 'timed out waiting'
+			info 'csv contents:'
+			oc get csv "${OPERATOR_PKG_NAME}" -n "${NAMESPACE}" -o yaml
+			exit 1
+		fi
+		sleep 10
+		continue
+	fi
+	info 'CSV successfully installed'
+	break
+done
+
+info 'deploy Quay'
+oc apply -n "${NAMESPACE}" -f "${QUAY_SAMPLE_PATH}"
+
+# wait for quay deployment to come up. the next checks will fail if the
+# resources don't exist.
+
+info 'waiting for quay deployment...'
+
+for n in {1..60}; do
+	deploy="$(oc get deploy skynet-quay-app -n "${NAMESPACE}" -o jsonpath='{.metadata.name}' 2> /dev/null)"
+	if [ ! "$deploy" = 'skynet-quay-app' ]; then
+		if [ "${n}" = "60" ]; then
+			error 'timed out waiting'
+			exit 1
+		fi
+		sleep 10
+		continue
+	fi
+	info 'quay deployment created'
+	break
+done
+
+# the reconcile loop recreates resources a few times before stabilising,
+# and the wait command isn't very smart (similar problems exist when waiting
+# for rollouts), so we just sleep a bit before waiting.
+# see https://github.com/kubernetes/kubectl/issues/1120 for details.
+info 'sleeping two minutes before waiting for pods...'
+sleep 120
+
+# the order below matters!
+# the mirror is the last to come up, so after the dependencies are ready
+# we wait for the mirror to be ready, then we check the quay app as it's
+# more likely to have stabilized after the mirror is up and running.
+info 'waiting for postgres pod...'
+oc -n "${NAMESPACE}" wait pods -l=quay-component=postgres --for=condition=Ready --timeout="${WAIT_TIMEOUT}"
+
+info 'waiting for redis pod...'
+oc -n "${NAMESPACE}" wait pods -l=quay-component=redis --for=condition=Ready --timeout="${WAIT_TIMEOUT}"
+
+info 'waiting for mirror pod...'
+oc -n "${NAMESPACE}" wait pods -l=quay-component=quay-mirror --for=condition=Ready --timeout="${WAIT_TIMEOUT}"
+
+info 'waiting for config-editor pod...'
+oc -n "${NAMESPACE}" wait pods -l=quay-component=quay-config-editor --for=condition=Ready --timeout="${WAIT_TIMEOUT}"
+
+info 'waiting for quay pod...'
+oc -n "${NAMESPACE}" wait pods -l=quay-component=quay-app --for=condition=Ready --timeout="${WAIT_TIMEOUT}"
+
+
+# manually check quay's health to ensure it can successfuly connect to its components.
+endpoint="$(oc -n "${NAMESPACE}" get quayregistry skynet -o jsonpath='{.status.registryEndpoint}')"
+result="$(curl -s -k "${endpoint}"/health/instance)"
+
+if [ "$(echo "${result}" | jq '.status_code')" != "200" ]; then
+	error 'quay health check did not return 200'
+	info "${result}"
+	info 'quay-app pod yaml:'
+	oc -n "${NAMESPACE}" logs -l=quay-component=quay-app
```

Comment:
can we dump here the quayregistry object as well?

Reply:
Good call, will do!

Reply:
done - see two lines above